scanpy/preprocessing/_pca.py

```diff
@@ -320,3 +370,44 @@ def rmatmat(x):
         'components': v,
     }
     return output
+
+
+def _handle_dask_ml_args(svd_solver: str, method: str) -> str:
+    method2args = {
+        'PCA': {'auto', 'full', 'tsqr', 'randomized'},
+        'IncrementalPCA': {'auto', 'full', 'tsqr', 'randomized'},
+        'TruncatedSVD': {'tsqr', 'randomized'},
+    }
+    method2default = {
+        'PCA': 'auto',
+        'IncrementalPCA': 'auto',
+        'TruncatedSVD': 'tsqr',
+    }
+
+    return _handle_x_args('dask_ml', svd_solver, method, method2args, method2default)
+
+
+def _handle_sklearn_args(svd_solver: str, method: str) -> str:
+    method2args = {
+        'PCA': {'auto', 'full', 'arpack', 'randomized'},
+        'TruncatedSVD': {'arpack', 'randomized'},
+        'PCA (with sparse input)': {'lobpcg', 'arpack'},
+    }
+    method2default = {
+        'PCA': 'arpack',
+        'TruncatedSVD': 'randomized',
+        'PCA (with sparse input)': 'arpack',
+    }
+
+    return _handle_x_args('sklearn', svd_solver, method, method2args, method2default)
+
+
+def _handle_x_args(lib, svd_solver, method, method2args, method2default):
+    changed = svd_solver not in method2args[method]
```

Comment:
`None` is not in there, so you still throw a warning by default.

Please add tests that check

1. no warning is thrown when the default (None) is specified
2. a warning is thrown when an inappropriate `svd_solver` is selected